modules/http4s/client/src/main/scala/Middleware.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2019 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez.http4s.client
+
+import cats.effect.{Resource, Sync}
+import natchez._
+import natchez.Tags._
+import org.http4s.client.Client
+import org.http4s.{Header, Headers}
+
+object Middleware {
+
+  def spannedClient[F[_]](client: Client[F], clientName: String)(implicit F: Sync[F], S: Span[F]): Client[F] = {
+    Client[F](request =>
+      for {
+        s      <- S.span(s"http4s-request-$clientName")
```

Comment:
I was under the impression Span isn't meant to be used like that (it's a data type and not a type class) and that `Trace[F].span` / `Trace[F].add` should be used to create new spans. Is that right?

Reply:
Ahh, I suppose the difference between `Trace` and `Span` is just not very clear to me. I can change this to use `Trace` instead if that's desired!